### PR TITLE
Logs index

### PR DESCRIPTION
**Pull request by miiu96**
- Added a new Elasticsearch index `logs` that will keep all the logs generated by transactions and smart contract results.
- Refactor how the NFT Create data is used. This information it's taken now from logs.

### CODE REVIEW

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: converters/tokenIdentifier.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package converters
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+)
+
+// ComputeTokenIdentifier will compute token identifier based on the token string and the nonce
```
[the remark this answers is not in the document]
Comment (miiu96) [reply converters/tokenIdentifier.go:9]: done.

File: converters/tokenMetaData.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import (
+	"github.com/ElrondNetwork/elastic-indexer-go/data"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-vm-common/data/esdt"
+)
+
+// PrepareTokenMetaData will prepare the token metadata in a friendly format for database
+func PrepareTokenMetaData(pubKeyConverter core.PubkeyConverter, esdtInfo *esdt.ESDigitalToken) *data.TokenMetaData {
+	if esdtInfo.TokenMetaData == nil {
+		return nil
```
[the remark this answers is not in the document]
Comment (miiu96) [reply converters/tokenMetaData.go:12]: I know what are you saying but in the case of a `fungible` token, `TokenMetaData` will be nil , no need for error here.

File: converters/tokenMetaData.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	}
+
+	creatorStr := ""
+	if esdtInfo.TokenMetaData.Creator != nil {
+		creatorStr = pubKeyConverter.Encode(esdtInfo.TokenMetaData.Creator)
+	}
+
+	return &data.TokenMetaData{
```
[the remark this answers is not in the document]
Comment (miiu96) [reply converters/tokenMetaData.go:20]: done.

File: data/altered.go [hunk 4]
```diff
@@ -46,12 +45,14 @@ func (aa *alteredAccounts) Add(key string, account *AlteredAccount) {
...
 	}
 
-	wasSender := false
+	senderCount := 0
 	for _, elem := range aa.altered[key] {
 		newElementIsTokenOperation := account.IsESDTOperation || account.IsNFTOperation
 		oldElementIsTokenOperation := elem.IsESDTOperation || elem.IsNFTOperation
 
-		wasSender = elem.IsSender || account.IsSender
+		if elem.IsSender || account.IsSender {
```
[the remark this answers is not in the document]
Comment (miiu96) [reply data/altered.go:53]: done.

File: process/logsevents/logsAndEventsProcessor.go [hunk 5]
```diff
@@ -1,39 +1,89 @@
...
 	return &logsAndEventsProcessor{
-		nftsProc: newNFTsProcessor(shardCoordinator, pubKeyConverter),
+		pubKeyConverter: pubKeyConverter,
+		nftsProc:        newNFTsProcessor(shardCoordinator, pubKeyConverter, marshalizer),
 	}, nil
 }
 
-// ProcessLogsAndEvents will process provided logs and events
-func (lep *logsAndEventsProcessor) ProcessLogsAndEvents(
+// ExtractDataFromLogsAndPutInAltered will extract information from provided logs and events and put in altered address
```
[the remark this answers is not in the document]
Comment (miiu96) [reply process/logsevents/logsAndEventsProcessor.go:43]: done.

File: process/logsevents/logsAndEventsProcessor.go [hunk 6]
```diff
@@ -1,39 +1,89 @@
...
+			continue
+		}
+
+		logs = append(logs, lep.prepareLogForDB(txHash, log))
+	}
+
+	return logs
+}
+
+func (lep *logsAndEventsProcessor) prepareLogForDB(id string, logHandler nodeData.LogHandler) *data.Logs {
```
[the remark this answers is not in the document]
Comment (miiu96) [reply process/logsevents/logsAndEventsProcessor.go:67]: done.

File: process/logsevents/nftsProcessor.go [hunk 7]
```diff
@@ -34,42 +43,63 @@ func newNFTsProcessor(
...
 	accounts data.AlteredAccountsHandler,
-) {
+	timestamp uint64,
+) (data.TokensHandler, tags.CountTags) {
 	if logsAndEvents == nil || accounts == nil {
-		return
+		return data.NewTokensInfo(), tags.NewTagsCount()
 	}
 
+	tagsCount := tags.NewTagsCount()
```
[the remark this answers is not in the document]
Comment (miiu96) [reply process/logsevents/nftsProcessor.go:52]: done.

File: process/logsevents/serialize.go [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/ElrondNetwork/elastic-indexer-go/data"
+)
+
+func (logsAndEventsProcessor) SerializeLogs(logs []*data.Logs) ([]*bytes.Buffer, error) {
```
[the remark this answers is not in the document]
Comment (miiu96) [reply process/logsevents/serialize.go:11]: done.

File: process/logsevents/logsAndEventsProcessor.go [hunk 9]
```diff
@@ -1,39 +1,89 @@
 package logsevents
 
 import (
+	"encoding/hex"
+
 	elasticIndexer "github.com/ElrondNetwork/elastic-indexer-go"
 	"github.com/ElrondNetwork/elastic-indexer-go/data"
+	"github.com/ElrondNetwork/elastic-indexer-go/process/tags"
 	"github.com/ElrondNetwork/elrond-go-logger/check"
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply process/logsevents/logsAndEventsProcessor.go:9]: not done yet

File: converters/tokenMetaData_test.go [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+func TestPrepareTokenMetaData(t *testing.T) {
+	t.Parallel()
+
+	require.Nil(t, PrepareTokenMetaData(nil, nil))
+	require.Nil(t, PrepareTokenMetaData(&mock.PubkeyConverterMock{}, nil))
+
+	require.Equal(t, &data.TokenMetaData{
```
Comment (iulianpascalau) [inline converters/tokenMetaData_test.go:18]: please add a supplemental variable expectedTokenMetaData. Also, please define a variable for the PrepareTokenMetaData call

File: converters/tags.go [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+const (
+	attributesSeparator = ";"
+	keyValuesSeparator  = ":"
+	valuesSeparator     = ","
+	tagsKey             = "tags"
+	metadataKey         = "metadata"
+)
+
+// ExtractTagsFromAttributes will extract tags from the attributes
+func ExtractTagsFromAttributes(attributes []byte) []string {
```
Comment (iulianpascalau) [inline converters/tags.go:14]: not a big fan for these static, exported functions

File: converters/tokenMetaData_test.go (hunk 10 above)
Comment (miiu96) [reply converters/tokenMetaData_test.go:18]: done.

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**